com.unity.render-pipelines.universal/ShaderLibrary/Lighting.hlsl

```diff
@@ -627,21 +627,146 @@ half3 SampleLightmap(float2 lightmapUV, half3 normalWS)
 #define SAMPLE_GI(lmName, shName, normalWSName) SampleSHPixel(shName, normalWSName)
 #endif
 
-half3 GlossyEnvironmentReflection(half3 reflectVector, half perceptualRoughness, half occlusion)
+half3 BoxProjectedCubemapDirection(half3 reflectionWS, half3 positionWS, float4 cubemapPositionWS, float4 boxMin, float4 boxMax)
+{
+    // Is this probe using box projection?
+    [branch] if (cubemapPositionWS.w > 0.0)
+    {
+        float3 boxMinMax = (reflectionWS > 0.0f) ? boxMax.xyz : boxMin.xyz;
+        half3 rbMinMax = half3(boxMinMax - positionWS) / reflectionWS;
+
+        half fa = half(min(min(rbMinMax.x, rbMinMax.y), rbMinMax.z));
+
+        half3 worldPos = half3(positionWS - cubemapPositionWS.xyz);
+
+        half3 result = worldPos + reflectionWS * fa;
+        return result;
+    }
+    else
+    {
+        return reflectionWS;
+    }
+}
+
+float CalculateProbeWeight(half3 positionWS, float4 probeBoxMin, float4 probeBoxMax)
+{
+    float blendDistance = probeBoxMax.w;
+    float3 weightDir = min(positionWS - probeBoxMin.xyz, probeBoxMax.xyz - positionWS) / blendDistance;
+    return saturate(min(weightDir.x, min(weightDir.y, weightDir.z)));
+}
+
+half CalculateProbeVolumeSqrMagnitude(float4 probeBoxMin, float4 probeBoxMax)
+{
+    half3 maxToMin = half3(probeBoxMax.xyz - probeBoxMin.xyz);
+    return dot(maxToMin, maxToMin);
+}
+
+half3 CalculateIrradianceFromReflectionProbes(half3 reflectVector, half3 positionWS, half perceptualRoughness)
+{
+    half probe0Volume = CalculateProbeVolumeSqrMagnitude(unity_SpecCube0_BoxMin, unity_SpecCube0_BoxMax);
+    half probe1Volume = CalculateProbeVolumeSqrMagnitude(unity_SpecCube1_BoxMin, unity_SpecCube1_BoxMax);
+
+    float volumeSign = sign(probe0Volume - probe1Volume);
+    float importanceSign = unity_SpecCube1_BoxMin.w;
+
+    // A probe is dominant if its importance is higher
+    // Or have equal importance but smaller volume
+    bool probe0Dominant = importanceSign > 0 || (importanceSign == 0 && volumeSign < 0);
+    bool probe1Dominant = importanceSign < 0 || (importanceSign == 0 && volumeSign > 0);
```

Comment:
`probe0Dominant` and `probe1Dominant` could be assigned directly to `unity_SpecCube1_BoxMin.w`.  Thereby avoiding the need to calculate volumes and determine the sign between the two volumes. Using the Mali offline compiler this would solve 3 arithmetic instructions. The trade-off here is that it'd bake the volume importance behavior into the component. It won't be possible for anyone to build blending that is only based on importance alone.

What is the preferred trade-off here?

Reply:
Did a performance test on mobile with Ernestas. Difference is not significant and not baking `w` gives more freedom. We will leave it as is.